TrigScint/src/TrigScint/Firmware/hitproducer_hw.cxx

```diff
@@ -0,0 +1,144 @@
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[15]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[16]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[17]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[18]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[19]
+
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[20]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[21]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[22]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[23]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[24]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[25]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[26]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[27]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[28]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[29]
+
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[30]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[31]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[32]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[33]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[34]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[35]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[36]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[37]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[38]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[39]
+
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[40]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[41]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[42]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[43]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[44]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[45]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[46]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[47]
+
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[48]
+#pragma HLS INTERFACE ap_fifo depth = 16 port = FIFO[49]
+
+#pragma HLS PIPELINE
+
+  /// Indices of first bin of each subrange
+  ap_uint<14> nbins_[5] = {0, 16, 36, 57, 64};
+
+  /// Charge lower limit of all the 16 subranges
+  ap_uint<14> edges_[17] = {0,     34,    158,    419,    517,   915,
+                            1910,  3990,  4780,   7960,   15900, 32600,
+                            38900, 64300, 128000, 261000, 350000};
+  /// sensitivity of the subranges (Total charge/no. of bins)
+  ap_uint<14> sense_[16] = {3,   6,   12,  25,   25,   50,   99,   198,
+                            198, 397, 794, 1587, 1587, 3174, 6349, 12700};
+
+  for (int i = 0; i < NHITS; i++) {
+    outHit[i].bID = -1;
+    outHit[i].mID = 0;
+    outHit[i].Time = 0;
+    outHit[i].Amp = 0;
+    ap_uint<14> word1 = FIFO[i][0];
+    ap_uint<14> word2 = FIFO[i][1];
+    ap_uint<14> word3 = FIFO[i][2];
+    ap_uint<14> word4 = FIFO[i][3];
+    ap_uint<14> word5 = FIFO[i][4];
+    ap_uint<14> word6 = FIFO[i][5];
```

Comment:
```
ap_uint<14> FIFO[NHITS][5]
```
i.e. the highest index you can have is `FIFO[i][4];` not the `FIFO[i][5];`

Reply:
Yes good catch. This was a big problem; I think I bungled the number just before submitting the pull request because right after I was having build issues. Thank you!